### PR TITLE
More small fixes and cleanups

### PR DESCRIPTION
**Pull request by oscargus**

### CODE REVIEW

File: src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java [hunk 1]
```diff
@@ -172,16 +180,10 @@ public boolean processQuery(String query, ImportInspector dialog, OutputPrinter
 
         } catch (MalformedURLException e) {
             LOGGER.warn("Bad URL", e);
-        } catch (ConnectException e) {
+        } catch (ConnectException | UnknownHostException e) {
             status.showMessage(Localization.lang("Connection to IEEEXplore failed"), DIALOG_TITLE,
```
Comment (koppor) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: Could we introduce the typical `"Connection to %0 failed", "IEEE"`? - See https://github.com/JabRef/jabref/pull/948. Maybe in a separate PR, but I just wanted that this is not forgotten.

File: src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java [hunk 2]
```diff
@@ -279,15 +281,15 @@ private BibEntry cleanup(BibEntry entry) {
             title = title.replaceAll("/[sS]pl ([^/]+)/", "\\$\\\\$1\\$");
             // Deal with subscripts and superscripts
             if (Globals.prefs.getBoolean(JabRefPreferences.USE_CONVERT_TO_EQUATION)) {
-                title = title.replaceAll("/sup ([^/]+)/", "\\$\\^\\{$1\\}\\$");
-                title = title.replaceAll("/sub ([^/]+)/", "\\$_\\{$1\\}\\$");
-                title = title.replaceAll("\\(sup\\)([^(]+)\\(/sup\\)", "\\$\\^\\{$1\\}\\$");
-                title = title.replaceAll("\\(sub\\)([^(]+)\\(/sub\\)", "\\_\\{$1\\}\\$");
+                title = title.replaceAll(SUP1, SUP_EQ_RESULT);
```
Comment (koppor) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: Would it be possible to use constant names, which differ more than one character. `P` vs. `B`? :innocent:

File: src/main/java/net/sf/jabref/model/database/BibDatabase.java [hunk 3]
```diff
@@ -83,8 +83,8 @@ public int getEntryCount() {
     /**
      * Checks if the database contains no entries.
      */
-    public boolean isEmpty() {
-        return entries.size() == 0;
+    public boolean hasNoEntries() {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/database/BibDatabase.java]: Why not keeping `isEmpty`, because that is more common than `hasNoEntries`?

**Comment by koppor** [comment]
Other than the micro comments: LGTM :+1:

File: src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java (hunk 2 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: I could change SUP to SUPER. :-)

File: src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: Changed it to (already existing) "Could not connect to %0". Waiting with removal until the automatic one is done. ;-)

File: src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java [hunk 4]
```diff
@@ -279,15 +281,15 @@ private BibEntry cleanup(BibEntry entry) {
...
             // Deal with subscripts and superscripts
             if (Globals.prefs.getBoolean(JabRefPreferences.USE_CONVERT_TO_EQUATION)) {
-                title = title.replaceAll("/sup ([^/]+)/", "\\$\\^\\{$1\\}\\$");
-                title = title.replaceAll("/sub ([^/]+)/", "\\$_\\{$1\\}\\$");
-                title = title.replaceAll("\\(sup\\)([^(]+)\\(/sup\\)", "\\$\\^\\{$1\\}\\$");
-                title = title.replaceAll("\\(sub\\)([^(]+)\\(/sub\\)", "\\_\\{$1\\}\\$");
+                title = title.replaceAll(SUPER_DETECTION_1, SUPER_EQ_RESULT);
+                title = title.replaceAll(SUB_DETECTION_1, SUB_EQ_RESULT);
+                title = title.replaceAll(SUPER_DETECTION_2, SUPER_EQ_RESULT);
+                title = title.replaceAll(SUB_DETECTION_2, SUB_EQ_RESULT);
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: For performance reasons, you should create Patterns as static variables and then use them to perform the replace all.

Comment (oscargus) [inline src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java]: Good idea! I'm actually wondering if these replacements (and quite a bit of the other cleanups) are useful any more, after the HTML-parsing has been replaced... Not sure how to fully determine it...

**Comment by simonharrer** [comment]
@koppor I am not sure whether they were micro comments or nano comments. Why would you say that they are micro comments?

**Pull request merged**